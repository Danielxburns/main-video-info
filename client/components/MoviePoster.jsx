import React from 'react';
import styled from 'styled-components';
// styled components
const PosterImage = styled.img`
  height: 40vh;
  width: 18vw;
`;


const MoviePoster = ({ poster }) => {
  // look into object destructuring
  return (
    <div>
      <PosterImage src={poster}/>
    </div>
  );
};

export default MoviePoster;