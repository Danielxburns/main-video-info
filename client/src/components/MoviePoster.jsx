import React from 'react';
import styled from 'styled-components';
// styled components
const PosterImage = styled.img`
  height: 38vh;
  width: 18vw;
  opacity: 1;
  transition: opacity .25s, transform .25s ease;
  :hover {
    transform: scale(1.09);
    opacity: 0.7;
  }
`;

const PosterDiv = styled.div`
  position: relative;
  display: inline-block;
  overflow: hidden;
  vertical-align: top;
`;


const MoviePoster = ({ poster }) => {
  // look into object destructuring
  return (
    <PosterDiv>
      <PosterImage src={poster}/>
    </PosterDiv>
  );
};

export {
  MoviePoster,
  PosterDiv,
  PosterImage
};