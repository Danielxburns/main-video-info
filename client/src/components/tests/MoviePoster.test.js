import React from 'react';
import { shallow, mount, render } from 'enzyme';
import {MoviePoster, PosterDiv, PosterImage} from '../MoviePoster';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import styled from 'styled-components';



describe("MoviePoster component", () => {
  it("should render", () => {
    const wrapper = shallow(<MoviePoster />);

    expect(wrapper.exists()).toBe(true);
  });

  it('applies correct styles to div containing poster', () => {
    const tree = renderer.create(<PosterDiv />).toJSON()
    expect(tree).toHaveStyleRule('position', 'relative');
  });
});