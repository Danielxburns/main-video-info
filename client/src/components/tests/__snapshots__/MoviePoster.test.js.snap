// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviePoster component should render correctly 1`] = `
.c0 {
  position: relative;
  display: inline-block;
  overflow: hidden;
  vertical-align: top;
}

.c1 {
  height: 38vh;
  width: 18vw;
  opacity: 1;
  -webkit-transition: opacity .25s,-webkit-transform .25s ease;
  -webkit-transition: opacity .25s,transform .25s ease;
  transition: opacity .25s,transform .25s ease;
}

.c1:hover {
  -webkit-transform: scale(1.09);
  -ms-transform: scale(1.09);
  transform: scale(1.09);
  opacity: 0.7;
}

<div
  className="c0"
>
  <img
    className="c1"
  />
</div>
`;
